stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/filepreview/MediaAttachmentPreviewHandler.kt

```diff
@@ -16,20 +16,18 @@ public class MediaAttachmentPreviewHandler(private val context: Context) : Attac
         val type = attachment.type
 
         if (assetUrl.isNullOrEmpty()) return false
-        if (mimeType.isNullOrEmpty()) return false
+
+        // Check if the base of Attachment.mimeType is audio or video
+        if (!mimeType.isNullOrEmpty() && (mimeType.contains(ModelType.attach_audio) || mimeType.contains(ModelType.attach_video))) return true
+
+        // If the previous check fails check Attachment.type for file type
         if (type.isNullOrEmpty()) return false
+        else if (type == ModelType.attach_audio || type == ModelType.attach_video) return true
 
-        val supportedMimeTypes = listOf(
-            MIME_TYPE_VIDEO,
-            MIME_TYPE_AUDIO,
-            // For compatibility with other client SDKs
-            MIME_SUBTYPE_MP4,
-            MIME_SUBTYPE_QUICKTIME
-        )
+        // Fallback in case we receive an incomplete mime type and both previous checks fail
+        val supportedMimeSubTypes = buildSubTypeList()
 
-        return supportedMimeTypes.any { mimeType.contains(it) } ||
-            type == ModelType.attach_audio ||
-            type == ModelType.attach_video
+        return supportedMimeSubTypes.any { mimeType?.contains(it) ?: false }
```

Comment:
Maybe do something like:

```
return if (mimeType == null) {
  false
} else {
  val supportedMimeSubTypes = buildSubTypeList()
  supportedMimeSubTypes.any { subtype -> subtype in mimeType }
}
```

This way we don't necessarily go through the entire list of subtypes to check with a null mime type that all values are null and false.

Reply:
Good suggestion, I've implemented it.